java/ql/lib/semmle/code/java/frameworks/ratpack/Ratpack.qll

```diff
@@ -83,5 +83,49 @@ private class RatpackModel extends SummaryModelCsv {
           "MultiValueMap;true;asMultimap;;;MapKey of Argument[-1];MapKey of ReturnValue;value",
           "MultiValueMap;true;asMultimap;;;MapValue of Argument[-1];MapValue of ReturnValue;value"
         ]
+    or
+    exists(string left, string right |
+      left = "Field[ratpack.func.Pair.left]" and
+      right = "Field[ratpack.func.Pair.right]"
+    |
+      row =
+        ["ratpack.util;", "ratpack.func;"] + "Pair;true;" +
+          [
+            "of;;;Argument[0];" + left + " of ReturnValue;value",
+            "of;;;Argument[1];" + right + " of ReturnValue;value",
+            "pair;;;Argument[0];" + left + " of ReturnValue;value",
+            "pair;;;Argument[1];" + right + " of ReturnValue;value",
+            "left;();;" + left + " of Argument[-1];ReturnValue;value",
+            "right;();;" + right + " of Argument[-1];ReturnValue;value",
+            "getLeft;;;" + left + " of Argument[-1];ReturnValue;value",
+            "getRight;;;" + right + " of Argument[-1];ReturnValue;value",
+            "left;(Object);;Argument[0];" + left + " of ReturnValue;value",
+            "left;(Object);;" + right + " of Argument[-1];" + right + " of ReturnValue;value",
+            "right;(Object);;Argument[0];" + right + " of ReturnValue;value",
+            "right;(Object);;" + left + " of Argument[-1];" + left + " of ReturnValue;value",
+            "pushLeft;(Object);;Argument[-1];" + right + " of ReturnValue;value",
+            "pushRight;(Object);;Argument[-1];" + left + " of ReturnValue;value",
```

Comment:
Missing: arg0 -> left/right of returnval

Reply:
You were indeed correct. Thank you for spotting this! I'm impressed with how thoroughly you check these PR's. I would have missed this as a reviewer.

Reply:
This should now be fixed. See below.